builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -208,6 +215,43 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 		Bucket: aws.String(d.Id()),
 	})
 	if err != nil {
+		// bucket may have things delete them
+		forceDestroy := d.Get("force_destroy").(bool); if (forceDestroy && strings.Contains(err.Error(),"BucketNotEmpty")) {
+			log.Printf("[DEBUG] S3 Bucket attempting to forceDestroy %+v", err)
+			
+			bucket := d.Get("bucket").(string)
+			resp, listErr := s3conn.ListObjects( 
+				&s3.ListObjectsInput {
+					Bucket: aws.String(bucket),
+				},
+			)
+
+			if (listErr != nil) {
```

Comment:
Go typically has you omit the parentheses here:

- https://golang.org/doc/effective_go.html